napari/__main__.py

```diff
@@ -211,9 +218,41 @@ def parse_sys_argv():
             unknown.append(args.paths.pop(len(args.paths) - idx - 1))
     kwargs = validate_unknown_args(unknown) if unknown else {}
 
+    if args.with_help_:
+        show_plugin_info()
+        sys.exit(0)
+
     return args, kwargs
 
 
+def show_plugin_info():
+    from .plugins import _initialize_plugins, _npe2, plugin_manager
+
+    # if a plugin widget has been requested, this will fail immediately
+    # if the requested plugin/widget is not available.
+    _initialize_plugins()
+    plugin_manager.discover_widgets()
+
+    shared_dict = {}
+    for _key, (_plugin_name, _widget_names) in _npe2.widget_iterator():
+        shared_dict.setdefault(_key, {}).setdefault('npe2', []).append(
+            (_plugin_name, _widget_names)
+        )
+    for _key, (_plugin_name, _widget_dict) in plugin_manager.iter_widgets():
+        shared_dict.setdefault(_key, []).setdefault(
```

Comment:
here is bug

Reply:
Removed printing of plugins from this PR -> Will move it to a separate one :)